src/plugins/ffmemless/plugin.c

```diff
@@ -578,10 +580,11 @@ static int ffm_play(struct ffm_effect_data *data, int play)
 	}
 
 #ifdef CACHE_EFFECTS
+	int16_t custom_data[CUSTOM_DATA_LEN] = {0, 0, 0};
 	if (play) {
 		data->cached_effect.id = -1;
 		if (data->cached_effect.type == FF_PERIODIC) {
-			int16_t custom_data[CUSTOM_DATA_LEN] = {data->customEffectId};
+			custom_data[0] = data->customEffectId;
 			data->cached_effect.u.periodic.custom_data = custom_data;
```

Comment:
Does that store a pointer to a variable allocated on stack (which soon goes out of scope) in a dynamically allocated structure or am I missing something?

Reply:
Thanks @monich for taking a look at this PR!

Yes, for the sake of upload, it does that. The data should be copied to kernel by the driver.

More or less what's happening here https://github.com/sonyxperiadev/vendor-qcom-opensource-vibrator/blob/aosp/LA.UM.7.1.r1/Vibrator.cpp#L80

Reply:
Actually, I did miss that it doesn't go out of scope anymore, before being accessed. That's better))